### PR TITLE
Merge latest changes from main

### LINKED ISSUE #1482: update Woodwork integration branch with latest from main
- Its been a while since the Woodwork integration has gotten the latest changes from `main` branch.
  - ex - it still tests python 3.9
- We should update the integration branch, and address the applicable conflicts (there is likely to be many)

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #1482 by merging the latest changes from the main branch and resolving conflicts that come up.

### CODE REVIEW

File: featuretools/entityset/entityset.py [hunk 1]
```diff
@@ -173,6 +173,21 @@ def __deepcopy__(self, memo):
...
+        '''String specifying the library used for the entity dataframes. Null if no entities'''
+        df_type = None
+
+        if self.entities:
+            if isinstance(self.entities[0].df, pd.DataFrame):
+                df_type = Library.PANDAS.value
+            elif isinstance(self.entities[0].df, dd.DataFrame):
+                df_type = Library.DASK.value
+            elif is_instance(self.entities[0].df, ks, 'DataFrame'):
+                df_type = Library.KOALAS.value
```
Comment (rwedge) [inline featuretools/entityset/entityset.py:187]: can you update this property to use the `self.dataframes` attribute instead of `self.entities`?

Comment (jeff-hernandez) [reply featuretools/entityset/entityset.py]: Yeah updated in 60350df50306d62acf3dbfea615986238a827845

**Review by rwedge — APPROVED** [review]
LGTM

File: featuretools/tests/conftest.py [hunk 2]
```diff
@@ -470,3 +470,46 @@ def koalas_entities():
...
+
+@pytest.fixture
+def pd_transform_es():
+    # Create dataframe
+    df = pd.DataFrame({'a': [14, 12, 10], 'b': [False, False, True],
+                       'b1': [True, True, False], 'b12': [4, 5, 6],
+                       'P': [10, 15, 12]})
+    es = ft.EntitySet(id='test')
+    # Add dataframe to entityset
+    es.entity_from_dataframe(entity_id='first', dataframe=df,
```
Comment (thehomebrewnerd) [inline featuretools/tests/conftest.py:488]: I think this is ok for now since I don't think we are using these fixtures in the integration yet, but these will need to be updated to `es.add_dataframe`...

File: featuretools/synthesis/get_valid_primitives.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+)
+from featuretools.synthesis.deep_feature_synthesis import DeepFeatureSynthesis
+from featuretools.synthesis.utils import (
+    _categorize_features,
+    get_unused_primitives
+)
+from featuretools.utils.gen_utils import Library
+
+
+def get_valid_primitives(entityset, target_entity, max_depth=2, selected_primitives=None):
```
Comment (thehomebrewnerd) [inline featuretools/synthesis/get_valid_primitives.py:14]: This can probably be a separate PR, but we need to update this function to remove references to  `entity`. Can you create an issue to capture this if you don't update it as part of this PR?

**Review by thehomebrewnerd — APPROVED** [review]
Just a couple small comments, but looks good.

File: featuretools/synthesis/get_valid_primitives.py (hunk 3 above)
Comment (jeff-hernandez) [reply featuretools/synthesis/get_valid_primitives.py:14]: Sounds good. I'll create issues for tracking

File: featuretools/tests/conftest.py (hunk 2 above)
Comment (jeff-hernandez) [reply featuretools/tests/conftest.py:488]: I'll create an issue for tracking

File: featuretools/synthesis/get_valid_primitives.py (hunk 3 above)
Comment (jeff-hernandez) [reply featuretools/synthesis/get_valid_primitives.py:14]: Filed #1494

File: featuretools/tests/conftest.py (hunk 2 above)
Comment (jeff-hernandez) [reply featuretools/tests/conftest.py:488]: Filed #1495
